pkg/autoscaler/statforwarder/forwarder_test.go

```diff
@@ -376,3 +385,88 @@ func TestForwarderSkipReconciling(t *testing.T) {
 		})
 	}
 }
+
+func TestProcess(t *testing.T) {
+	t.Parallel()
+	ctx, cancel, _ := rtesting.SetupFakeContextWithCancel(t)
+	kubeClient := fakekubeclient.Get(ctx)
+	endpoints := fakeendpointsinformer.Get(ctx)
+	service := fakeserviceinformer.Get(ctx)
+	lease := fakeleaseinformer.Get(ctx)
+
+	waitInformers, err := controller.RunInformers(
+		ctx.Done(), endpoints.Informer(), service.Informer(), lease.Informer())
+	if err != nil {
+		t.Fatal("Failed to start informers:", err)
+	}
+
+	t.Cleanup(func() {
+		cancel()
+		waitInformers()
+	})
+
+	acceptCount := 0
+	accept := func(sm asmetrics.StatMessage) {
+		acceptCount++
+	}
+	f := New(ctx, zap.NewNop().Sugar(), kubeClient, testIP1, hash.NewBucketSet(sets.NewString(bucket1, bucket2)), accept)
+
+	stat1 := asmetrics.StatMessage{
+		Key: types.NamespacedName{
+			Namespace: testNs,
+			Name:      "succulent", // Mapped to bucket1
+		},
+	}
+	stat2 := asmetrics.StatMessage{
+		Key: types.NamespacedName{
+			Namespace: testNs,
+			Name:      "plant", // Mapped to bucket2
+		},
+	}
+
+	// A Forward without any leadership information should process without error.
+	f.Process(stat1)
+
+	kubeClient.CoordinationV1().Leases(testNs).Create(testLease)
+	lease.Informer().GetIndexer().Add(testLease)
+	anotherLease := &coordinationv1.Lease{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      bucket2,
+			Namespace: testNs,
+		},
+		Spec: coordinationv1.LeaseSpec{
+			HolderIdentity: &testIP2,
+		},
+	}
+	kubeClient.CoordinationV1().Leases(testNs).Create(anotherLease)
+	lease.Informer().GetIndexer().Add(anotherLease)
+
+	// Wait for the forwarder to become the leader for bucket1.
+	if err := wait.PollImmediate(100*time.Millisecond, 2*time.Second, func() (bool, error) {
+		p1, ok1 := f.getProcessor(bucket1)
+		p2, ok2 := f.getProcessor(bucket2)
+		return ok1 && ok2 && p1.holder == testIP1 && p2.holder == testIP2 && p1.conn == nil && p2.conn != nil, nil
+	}); err != nil {
+		t.Fatalf("Timeout waiting f.processors got updated")
+	}
+
+	forwardCount := 0
+	// Override the proc so we do not actually send via WebSocket.
+	f.processors[bucket2].proc = func(sm asmetrics.StatMessage) {
+		forwardCount++
+	}
+
+	f.Process(stat1)
+	f.Process(stat2)
+	f.Process(stat2)
+
+	if got, want := acceptCount, 1; got != want {
+		t.Errorf("accpetCount = %d, want = %d", got, want)
+	}
+	if got, want := forwardCount, 2; got != want {
+		t.Errorf("forwardCount = %d, want = %d", got, want)
+	}
+
+	// Check Cancel is called without error.
```

Comment:
comment says we check, but I don't think we're checking?

Reply:
Reworded the comment. I intend to make sure that this can be called without crash (by nil point).